packages/expo-crypto/android/src/main/java/expo/modules/crypto/CryptoModule.kt

```diff
@@ -0,0 +1,46 @@
+package expo.modules.crypto
+
+import android.content.Context
+import android.util.Base64
+
+import expo.modules.core.ExportedModule
+import expo.modules.core.Promise
+import expo.modules.core.interfaces.ExpoMethod
+
+import java.security.MessageDigest
+import java.security.NoSuchAlgorithmException
+
+class CryptoModule(context: Context) : ExportedModule(context) {
+  override fun getName() = "ExpoCrypto"
+
+  @ExpoMethod
+  fun digestStringAsync(algorithm: String, data: String, options: Map<String, Any?>, promise: Promise) {
+    val encoding = options["encoding"] as String?
```

Comment:
Maybe we should do some checking here instead of casting: 🤔 
```suggestion
    val encoding = options["encoding"].let {
      return@let if (it is String) {
        it
      } else {
        promise.reject("ERR_CRYPTO_DIGEST", "Encoding should be a string")
        null
      }
    } ?: return
```